javascript/react/src/components/Codebook/CodebookTableEditor.tsx

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { Field } from '@datashaper/schema'
+import styled from '@essex/styled-components'
+import { memo, useCallback } from 'react'
+
+import { CodebookFieldEditor } from './CodebookFieldEditor.js'
+import type { CodebookTableEditorProps } from './CodebookTableEditor.types.js'
+
+export const CodebookTableEditor: React.FC<CodebookTableEditorProps> = memo(
+	function CodebookTableEditor({
+		fields,
+		onChange,
+		showInlineLabel,
+		showFields,
+	}) {
+		const onChangeField = useCallback(
+			(newField: Field) => {
+				const fieldIndex = fields.findIndex(x => x.name === newField.name)
+				onChange([
+					...fields.slice(0, fieldIndex),
+					newField,
+					...fields.slice(fieldIndex + 1),
+				])
+			},
+			[onChange, fields],
+		)
+
+		return (
+			<Container>
+				{fields.map((f: Field, index: number) => {
```

Comment:
You could create the onChange handler here and get the index property from the `map` operation, removing the need to find it above.